modules/src/ics02_client/client.rs

```diff
@@ -9,12 +11,20 @@ pub struct RawClientConnections {
     pub connections: ::std::vec::Vec<String>,
 }
 
-impl TryFromRaw for Vec<String> {
+impl TryFromRaw for Vec<ConnectionId> {
     type RawType = RawClientConnections;
     type Error = anomaly::Error<Kind>;
     fn try_from(value: RawClientConnections) -> Result<Self, Self::Error> {
         if !value.connections.is_empty() {
-            Ok(value.connections)
+            let mut connections: Vec<ConnectionId> = vec![];
```

Comment:
Can this be done with a `.map`?

Reply:
it could, but because this can return an error (in case a connection id is not valid, eg. longer than max characters) this is implementation is not too bad. I had the assumption that for some reason there are multiple connections and one of them is not valid then an error is returned right away. There might be patterns with map you can accomplish that with things like ok_or_else and fold but sometimes I think this is not as intuitive as some explicit code sometimes. If you think it's worth it I can refactor it.

Reply:
I'll just resolve the conversation for now. If we think we should refactor we can open a separate issue for that.